src/Build/BackEnd/Components/RequestBuilder/TargetBuilder.cs

```diff
@@ -791,12 +791,13 @@ private void ComputeAfterTargetFailures(string[] targetNames)
                             if (!resultsWithAfterTargets.TryGetValue(afterTarget.TargetName, out bool afterTargetFailed))
                             {
                                 _buildResult.ResultsByTarget.TryGetValue(afterTarget.TargetName, out TargetResult result);
-                                afterTargetFailed = result != null && result.ResultCode == TargetResultCode.Failure && !result.TargetFailureDoesntCauseBuildFailure;
+                                afterTargetFailed = result?.ResultCode == TargetResultCode.Failure && !result.TargetFailureDoesntCauseBuildFailure;
 
                                 resultsWithAfterTargets[afterTarget.TargetName] = afterTargetFailed;
                                 if (afterTargetFailed)
                                 {
                                     _buildResult.ResultsByTarget[targetName].AfterTargetsHaveFailed = true;
+                                    break;
```

Comment:
This only breaks the innermost foreach, not the while as well right? You could empty the queue or use a bool.

Reply:
Could also use `return`, since there isn't anything after that.

Reply:
I think returning would be incorrect, as the outermost foreach over `targetNames` needs to explore whether each target's aftertargets closure contains a failing target.

Reply:
Fixed, thank you!

Reply:
You're right. Too many comments—I missed the third loop.